src/DDTrace/Http/Urls.php

```diff
@@ -2,12 +2,37 @@
 
 namespace DDTrace\Http;
 
-
 /**
  * A utility class that provides methods to work on urls
  */
 class Urls
 {
+    private static $defaultPatterns = [
+        // UUID's
+        '|\b([0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89AB][0-9a-f]{3}-[0-9a-f]{12})\b|i',
+        // 16-512 bit hex hashes
+        '|\b([0-9a-f]{4,128})\b|i',
+        // int's
+        '|\b([0-9]+)\b|',
+    ];
+
+    private $replacementPatterns = [];
+
+    /**
+     * Inject URL replacement patterns using '*' and '$*' as wildcards
+     * The '*' wildcard will match one or more characters to be replaced with '?'
+     * The '$*' wildcard will match one or more characters without replacement
+     *
+     * @param string[] $patternsWithWildcards
+     */
+    public function __construct(array $patternsWithWildcards = [])
+    {
+        $this->replacementPatterns = array_map(
```

Comment:
[Optional]: this is a matter of styles and this comment is totally optional. I love to `Ctrl+click` around to see who is using a method. While I know that UTs would detect something wrong here, I still prefer the plain old `for` loop. This is personal here :) don't hate me 😄

Reply:
Lol - I don't mind refactoring this to a `foreach` loop since we aren't able to support `ClassName::class`. :)